software/contrib/turing_machine.py

```diff
@@ -199,8 +215,12 @@ def __init__(self, bit_count=DEFAULT_BIT_COUNT, max_output_voltage=MAX_OUTPUT_VO
         self.kb2 = (
             KnobBank.builder(k2)
             .with_disabled_knob()
-            .with_locked_knob("scale", initial_value=0)
-            .with_locked_knob("length", initial_value=0)
+            .with_locked_knob("scale", initial_percentage_value=initial_scale_percent)
+            .with_locked_knob(
+                "length",
+                initial_percentage_value=(self.LENGTH_CHOICES.index(initial_length) * 2 + 1)
```

Comment:
So I did start going down this road earlier in the development of `LockableKnob` but I backed off of it because of the flexibility that the `Knob` api allows. Specifically, the api allows the client to read the knob for any number of choices at any time. So given that, saving the value for 'choice 2' felt weird to me.

That said, thinking about your suggestion more and looking at this code here, I think that you are right and we could make a 'choices' api for lockable knob that would make sense and be way cleaner than this mess.

I think that this change would be an improvement beyond the scope of this PR however, would you be ok with it coming in a followup PR?